Flaky Tests: AbstractTestArrowFederationNativeQueries

<!--- Provide a general summary of the issue in the Title above -->
<!--- Look through existing open and closed issues to see if someone has reported the issue before -->

Failed even after 3 retries. 

One of the executions: https://github.com/prestodb/presto/actions/runs/29856469259/job/88862453301#step:12:1


```

2026-07-22T08:57:07.9780239Z [INFO] Results:
2026-07-22T08:57:07.9780606Z [INFO] 
2026-07-22T08:57:07.9780947Z [ERROR] Failures: 
2026-07-22T08:57:07.9785248Z [ERROR]   TestArrowFederationNativeQueriesCassandra>AbstractTestArrowFederationNativeQueries.testDuplicateUnnestRows:445->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)
2026-07-22T08:57:07.9792348Z [ERROR]   TestArrowFederationNativeQueriesCassandra>AbstractTestArrowFederationNativeQueries.testSetUnionIndeterminateRows:467->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT c1, c2 from (SELECT set_union(x) as agg_result from (SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))] x UNION ALL SELECT ARRAY[null, CAST(row(1, null) AS ROW(INTEGER, INTEGER))] UNION ALL SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) CROSS JOIN unnest(agg_result) as r(c1, c2)
2026-07-22T08:57:07.9797178Z [ERROR]   TestArrowFederationNativeQueriesMongo>AbstractTestArrowFederationNativeQueries.testDuplicateUnnestRows:445->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)
2026-07-22T08:57:07.9802898Z [ERROR]   TestArrowFederationNativeQueriesMongo>AbstractTestArrowFederationNativeQueries.testSetUnionIndeterminateRows:467->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT c1, c2 from (SELECT set_union(x) as agg_result from (SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))] x UNION ALL SELECT ARRAY[null, CAST(row(1, null) AS ROW(INTEGER, INTEGER))] UNION ALL SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) CROSS JOIN unnest(agg_result) as r(c1, c2)
2026-07-22T08:57:07.9808091Z [ERROR]   TestArrowFederationNativeQueriesMySql>AbstractTestArrowFederationNativeQueries.testDuplicateUnnestRows:445->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)
2026-07-22T08:57:07.9813311Z [ERROR]   TestArrowFederationNativeQueriesMySql>AbstractTestArrowFederationNativeQueries.testSetUnionIndeterminateRows:467->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT c1, c2 from (SELECT set_union(x) as agg_result from (SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))] x UNION ALL SELECT ARRAY[null, CAST(row(1, null) AS ROW(INTEGER, INTEGER))] UNION ALL SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) CROSS JOIN unnest(agg_result) as r(c1, c2)
2026-07-22T08:57:07.9818099Z [ERROR]   TestArrowFederationNativeQueriesPostgres>AbstractTestArrowFederationNativeQueries.testDuplicateUnnestRows:445->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)
2026-07-22T08:57:07.9823501Z [ERROR]   TestArrowFederationNativeQueriesPostgres>AbstractTestArrowFederationNativeQueries.testSetUnionIndeterminateRows:467->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT c1, c2 from (SELECT set_union(x) as agg_result from (SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))] x UNION ALL SELECT ARRAY[null, CAST(row(1, null) AS ROW(INTEGER, INTEGER))] UNION ALL SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) CROSS JOIN unnest(agg_result) as r(c1, c2)
2026-07-22T08:57:07.9828381Z [ERROR]   TestArrowFederationNativeQueriesRedis>AbstractTestArrowFederationNativeQueries.testDuplicateUnnestRows:445->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)
2026-07-22T08:57:07.9833417Z [ERROR]   TestArrowFederationNativeQueriesRedis>AbstractTestArrowFederationNativeQueries.testSetUnionIndeterminateRows:467->AbstractTestQueryFramework.assertQueryFails:336 Expected exception message 'unnestNames_.size() == expectedUnnestNames (2 vs. 1) UnnestNode requires one name per array column and two per map column' to match 'Field not found: field(?:_\d+)?. Available fields are: field.*' for query: SELECT c1, c2 from (SELECT set_union(x) as agg_result from (SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))] x UNION ALL SELECT ARRAY[null, CAST(row(1, null) AS ROW(INTEGER, INTEGER))] UNION ALL SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) CROSS JOIN unnest(agg_result) as r(c1, c2)
2026-07-22T08:57:07.9836321Z [INFO] 
2026-07-22T08:57:07.9836617Z [ERROR] Tests run: 2665, Failures: 10, Errors: 0, Skipped: 50
2026-07-22T08:57:07.9836953Z [INFO] 
2026-07-22T08:57:07.9856754Z [INFO] ------------------------------------------------------------------------
2026-07-22T08:57:07.9857437Z [INFO] BUILD FAILURE
2026-07-22T08:57:07.9858901Z [INFO] ------------------------------------------------------------------------
2026-07-22T08:57:07.9890009Z [INFO] Total time:  01:05 h (Wall Clock)
2026-07-22T08:57:07.9913112Z [INFO] Finished at: 2026-07-22T02:57:07-06:00
2026-07-22T08:57:07.9913730Z [INFO] ------------------------------------------------------------------------
2026-07-22T08:57:07.9926526Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project presto-flight-shim: There are test failures.
2026-07-22T08:57:07.9927792Z [ERROR] 
2026-07-22T08:57:07.9929130Z [ERROR] Please refer to /__w/presto/presto/presto-flight-shim/target/surefire-reports for the individual test results.
2026-07-22T08:57:07.9930164Z [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2026-07-22T08:57:07.9930798Z [ERROR] -> [Help 1]
2026-07-22T08:57:07.9931007Z [ERROR] 
2026-07-22T08:57:07.9931359Z [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2026-07-22T08:57:07.9931895Z [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2026-07-22T08:57:07.9932257Z [ERROR] 
2026-07-22T08:57:07.9932702Z [ERROR] For more information about the errors and possible solutions, please read the following articles:
2026-07-22T08:57:07.9933621Z [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
2026-07-22T08:57:08.0499759Z ##[error]Process completed with exit code 1.

```

--- a/presto-flight-shim/src/test/java/com/facebook/presto/flightshim/AbstractTestArrowFederationNativeQueries.java
+++ b/presto-flight-shim/src/test/java/com/facebook/presto/flightshim/AbstractTestArrowFederationNativeQueries.java
@@ -39,7 +39,7 @@ import static com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder;
 public abstract class AbstractTestArrowFederationNativeQueries
         extends AbstractTestDistributedQueries
 {
-    private final @Language("RegExp") String unnestRowsInvalidFieldError = "Field not found: field(?:_\\d+)?. Available fields are: field.*";
+    private final @Language("RegExp") String unnestRowsError = ".*UnnestNode requires one name per array column and two per map column.*";
 
     protected BufferAllocator allocator;
     protected FlightShimProducer producer;
@@ -443,18 +443,18 @@ public abstract class AbstractTestArrowFederationNativeQueries
     public void testDuplicateUnnestRows()
     {
         assertQueryFails("SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
         assertQueryFails("SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)], ARRAY[row(10, 13, 15), row(23, 25, 20)]) AS r(r1, r2, r3, r4, r5, r6, r7)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
         assertQueryFails("SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) WITH ORDINALITY AS r(r1, r2, r3, r4, ord)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
         assertQueryFails("SELECT * from (select * FROM (values 1) as t(k)) CROSS JOIN unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)], ARRAY[row(10, 13, 15), row(23, 25, 20)]) WITH ORDINALITY AS r(r1, r2, r3, r4, r5, r6, r7, ord)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
 
         assertQueryFails("SELECT * from unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) AS r(r1, r2, r3, r4)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
         assertQueryFails("SELECT * from unnest(ARRAY[row(2, 3), row(3, 5)], ARRAY[row(2, 3), row(3, 5)]) WITH ORDINALITY AS r(r1, r2, r3, r4, ord)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
     }
 
     /// Presto C++ only supports legacy unnest and this test relies on non-legacy behavior of unnest operator for
@@ -471,7 +471,7 @@ public abstract class AbstractTestArrowFederationNativeQueries
                         "UNION ALL " +
                         "SELECT ARRAY[CAST(row(null, 2) AS ROW(INTEGER, INTEGER))])) " +
                         "CROSS JOIN unnest(agg_result) as r(c1, c2)",
-                unnestRowsInvalidFieldError, true);
+                unnestRowsError, true);
     }
 
     /// The integer overflow error message differs in Presto and Velox.

